test/src/main/java/org/apache/accumulo/test/functional/CompactionIT.java

```diff
@@ -280,6 +280,27 @@ public void testUserCompactionCancellation() throws Exception {
     }
   }
 
+  @Test
+  public void testErrorDuringUserCompaction() throws Exception {
+    final String table1 = this.getUniqueNames(1)[0];
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(table1);
+      try (BatchWriter bw = client.createBatchWriter(table1)) {
+        for (int i = 1; i <= MAX_DATA; i++) {
+          Mutation m = new Mutation(Integer.toString(i));
+          m.put("cf", "cq", new Value());
+          bw.addMutation(m);
+          bw.flush();
+          client.tableOperations().flush(table1, null, null, true);
+        }
+      }
```

Comment:
Do you need to flush the table for every mutation? Are you trying to create separate files for each mutation? If so, this could use a comment. Otherwise, this can be simplified:

```suggestion
        }
      }
      client.tableOperations().flush(table1, null, null, true);
```

Reply:
I just copied the test above and modified to add the iterator. I presume that it's flushing so frequently to create separate files for the compaction tests. Otherwise, it would likely just create one file and the compaction test would really do nothing.

Reply:
In that case, this PR looks good to me, but I still think a comment that says `// create separate files` would be helpful inline. If you copied from another test, then putting it there would be nice, too.

Reply:
Comments added in fa410ed